doc/examples/with_widgets_overlay/lib/main.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:flutter/material.dart';
+import 'package:flame/game.dart';
+import 'package:flame/gestures.dart';
+
+class ExampleGame extends Game with HasWidgetsOverlay, TapDetector {
+  bool isPaused = false;
+
+  @override
+  void update(double dt) {}
+
+  @override
+  void render(Canvas canvas) {
+    canvas.drawRect(const Rect.fromLTWH(100, 100, 100, 100),
+        Paint()..color = const Color(0xFFFFFFFF));
+  }
+
+  @override
+  void onTap() {
+    if (isPaused) {
+      removeWidgetOverlay("PauseMenu");
+      isPaused = false;
+    } else {
+      addWidgetOverlay(
+          "PauseMenu",
+          Center(
```

Comment:
My concern here is that, by design, this API doesn't allow the widgets have access to a build context, making it impossible to be used with things such as themes, inherited widgets (provider) and even router.
If the intention is to allow widgets here be used just like in any flutter screen, we have to give access to the build context.

Reply:
I agree and disagree with you. Sure we should have the access to the context here, but the problem is that is that Flame don't give the access to the BuildContext out of the box, that is something needed even before this API.

So I don't this is a matter of this API or PR, and is something that we should handle, but in another PR.

Reply:
agree this can be a followup, also I wonder how really important are these more complex features in the context of a simple in-game dialog.

Reply:
Exactly, and on the worst case scenario, there are tons of ways for the Game instances receive a context, so that can still be used if truly needed.